drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+            /* clip to 16 bits */
+            req_spads = (req_spads > 0xffff) ? 0xffff : req_spads;
+
+            /* override DSS config */
+            EXEC_RET(_write_word(dev, VL53L1X_DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT,
+                                      req_spads));
+            /*
+             * VL53L1X_DSS_CONFIG__ROI_MODE_CONTROL should already be set to
+             * REQUESTED_EFFFECTIVE_SPADS
+             */
+
+             return VL53L1X_OK;
+        }
+    }
+
+    /*
+     * If we reached this point, it means something above would have resulted
+     * in a divide by zero. We want to gracefully set a spad target, not just
+     * exit with an error
+     */
+    /* set target to mid point */
+    EXEC_RET(_write_word(dev, VL53L1X_DSS_CONFIG__MANUAL_EFFECTIVE_SPADS_SELECT,
+                              0x8000));
+    return VL53L1X_OK;
+}
+
+/* Copyrighted POLOLU / ST API code end */
+
+#define VL53L1X_BUFSIZ   32
+static uint8_t _buffer[VL53L1X_BUFSIZ] = {};
+
+static int _read_byte(const vl53l1x_t *dev, uint16_t index, uint8_t *data)
+{
+    return _read (dev, index, data, 1);
+}
+
+static int _read_word(const vl53l1x_t *dev, uint16_t index, uint16_t *data)
+{
+    uint8_t bytes[2];
+    EXEC_RET(_read (dev, index, bytes, 2));
+    *data = (bytes[0] << 8) + (bytes[1]);
+
+    return VL53L1X_OK;
+}
+
+static int _read(const vl53l1x_t *dev, uint16_t index, uint8_t *pdata, uint32_t len)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(pdata != NULL);
+    DEBUG_DEV("index=%04x pdata=%p len=%u", dev, index, pdata, len);
+
+    /* acquire the I2C device */
+    if (i2c_acquire(dev->params.i2c_dev)) {
+        DEBUG_DEV("could not aquire I2C bus", dev);
+        return VL53L1X_ERROR_I2C;
+    }
+
+    /* read blockwise with a maximum size of VL53L1X_BUFSIZ */
+    while (len) {
+
+        uint8_t bytes[2] = { (index >> 8) & 0xff, index & 0xff };
+
+        /* write the register index for the block */
+        if (i2c_write_bytes(dev->params.i2c_dev, VL53L1X_I2C_ADDRESS, bytes, 2, 0) != 0) {
+            i2c_release(dev->params.i2c_dev);
+            return VL53L1X_ERROR_I2C;
+        }
+
+        uint8_t cnt = (len < VL53L1X_BUFSIZ) ? len : VL53L1X_BUFSIZ;
+
+        if (i2c_read_bytes(dev->params.i2c_dev, VL53L1X_I2C_ADDRESS, _buffer, cnt, 0) != 0) {
+            i2c_release(dev->params.i2c_dev);
+            return VL53L1X_ERROR_I2C;
+        }
+
+        memcpy(pdata, _buffer, cnt);
+        len -= cnt;
+        pdata += cnt;
+        index += cnt;
+    }
+
+    /* release the I2C device */
+    i2c_release(dev->params.i2c_dev);
+
+    return VL53L1X_OK;
+}
+
+static int _write_byte(const vl53l1x_t *dev, uint16_t index, uint8_t data)
+{
+    return _write (dev, index, &data, 1);
+}
+
+static int _write_word(const vl53l1x_t *dev, uint16_t index, uint16_t data)
+{
+    uint8_t bytes[2] = { (data >> 8) & 0xff, data & 0xff };
+    return _write (dev, index, (uint8_t*)&bytes, 2);
```

Comment:
```suggestion
    return _write(dev, index, bytes, 2);
```

`&bytes` is a of type `uint8_t **`, so pointer to a pointer to `uint8_t`. So previously this would have written two bytes of the address of `bytes`.

Reply:
Fixed with commit https://github.com/RIOT-OS/RIOT/pull/10363/commits/71fd9adb7b92b9a63b3b84f731b7cdec408065c3.